src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,108 @@
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    )
+        public virtual initializer
+    {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token) external onlyWhenNotPaused(PAUSED_NEW_STRATEGIES) returns (IStrategy newStrategy) {
+        require(tokenStrategy[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token");
+        IStrategy strategy = IStrategy(address(
+            new BeaconProxy(
+                address(strategyBeacon),
+                abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+            )
+        ));
+        _setStrategyForToken(token, strategy);
+        IStrategy[] memory strategiesToWhitelist = new IStrategy[](1);
+        bool[] memory thirdPartyTransfersForbiddenValues = new bool[](1);
+        strategiesToWhitelist[0] = strategy;
+        thirdPartyTransfersForbiddenValues[0] = false;
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
+        return strategy;
+    }
+
+    /** 
+     * @notice Owner-only function to pass through a call to `StrategyManager.addStrategiesToDepositWhitelist`
+     * @dev Also adds the `strategiesToWhitelist` to the `tokenStrategy` mapping
+     */
+    function whitelistStrategies(
+        IStrategy[] calldata strategiesToWhitelist,
+        bool[] calldata thirdPartyTransfersForbiddenValues
+    ) external onlyOwner {
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
+        for (uint256 i = 0; i < strategiesToWhitelist.length; ++i) {
+            IERC20 underlyingToken = strategiesToWhitelist[i].underlyingToken();
+            _setStrategyForToken(underlyingToken, strategiesToWhitelist[i]);
+        }
+    }
+
+    // @notice Owner-only function to add (existing) Strategy contracts to the `tokenStrategy` mapping
+    function editTokenStrategiesMapping(
```

Comment:
This seems danger - we want to be able to guarantee what strategies are added to that mapping, their implementations, their proper beacon proxies, etc.

Reply:
that's fair; like many owner-only functions, this could be misused.
the severity of the misuse depends on what value you place on the storage -- I was viewing it primarily as a deployment-duplication-prevention mechanism, but it could potentially serve more of a purpose

Reply:
function removed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/b88cdb5617d8ba51b15b6472f655d8043d1929ee